Overview.bs

```diff
@@ -212,39 +212,131 @@ Opt-In Mechanism {#opt-in}
 
 <em>This section is general to both IFT methods.</em>
 
-The collection of IFT technologies utilizes a single shared opt-in mechanism. Each method does not use its own opt-in mechanism; instead, the webpage opts into the IFT technologies as a whole, and the browser and server negotiate to decide which specific method will be employed.
+Webpage's can choose to opt-in to either patch subset or range request incremental transfer for a font
+via the use of a CSS font tech keyword ([[css-fonts-4#font-tech-definitions]]) inside the
+''@font-face'' block.
 
-The opt-in mechanism is the <code>incremental</code> keyword inside the ''@font-face'' block. Websites specify this keyword inside their <code>@font-face</code> block, and the browser is then responsible for using IFT technologies, and for negotiating with the server to determine which specific IFT method to use.
+There are three tech keywords available:
+
+*  <code>incremental-patch</code>: Load the font incrementally using the patch subset method.
+*  <code>incremental-range</code>: Load the font incrementally using the range request method.
+*  <code>incremental-auto</code>:  Load the font incrementally. The particular method to use is
+    determined by <a href="#method-negotiation">Method Negotiation</a>.
 
 <div class=example>
-The use of the <code>incremental</code> keyword in this CSS rule indicates to the browser they should use IFT.
+The use of the <code>incremental-patch</code> keyword in this CSS rule indicates to the browser they
+should use the patch subset IFT method to load the font.
 <pre>
 @font-face {
     font-family: "MyCoolWebFont";
-    src: url("MyCoolWebFont.otf") tech(incremental);
+    src: url("MyCoolWebFont.otf") tech(incremental-patch);
 }
 </pre>
 </div>
 
 Note: Each individual <code>@font-face</code> block may or may not opt-in to IFT. This is due to the variety of ways fonts are used on web pages. Authors have control over which fonts they want to use this technology with, and which they do not.
 
+Note: the use of <code>incremental-auto</code> may incur a CORS preflight request for the initial
+request of method negotiation, as the initial request sets a custom header.
+
 IFT Method Selection {#method-selection}
 ========================================
 
 <em>This section is general to both IFT methods.</em>
 
-When a page has indicated that a particular font should utilize IFT technology, the browser must determine which method to use. Different browsers may support different IFT methods, and different servers may support different IFT methods, so a negotation occurs as such:
+The client selects the IFT method to utilize for a font load using the following procedure:
+
+1. If the content has specified which method to use, such as via the
+    <a href="#opt-in">opt-in mechanism</a>, then that method should be used if the client supports
+    it.
+
+2. If the client does not support the specified method then it should fallback to to loading
```

Comment:
So (just to be clear) if the CSS specified `incremental-patch` and the client does not support it (but does support `incremental-range`); and the client has not contacted the server yet; then it must assume the server does not support `incremental-range` and just request the whole font? (And the same if the CSS says `incremental-range` and the client does not support it).

It seems odd to have this as step 2, and then step 3 is "if the client does not know the server capabilities" which implies at step 2 it *does* know.

What am I not getting, here?

Reply:
I'll reword this. There are basically two cases:

content specifies incremental-patch or incremental-range which signals what the server supports (or in the non browser based case the client through some means knows which method the server supports). In that case proceed with the indicated method if supported.

Or content specifies incremental-auto (or this is a non browser based case where the server support is unknown in advance). Then the client should use negotiation to sort it out.

Reply:
Got it.